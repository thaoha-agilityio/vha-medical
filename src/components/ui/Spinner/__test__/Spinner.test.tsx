import { render } from '@testing-library/react';

// Component
import { Spinner } from '..';
import { SpinnerProps } from '@nextui-org/react';

describe('Spinner Component', () => {
  const setup = (props: SpinnerProps) => render(<Spinner {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render Spinner component correctly without props', () => {
    const { asFragment } = setup({});

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Spinner component correctly with props', () => {
    const { asFragment } = setup({
      classNames: {
        circle1: '',
        circle2: '',
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
