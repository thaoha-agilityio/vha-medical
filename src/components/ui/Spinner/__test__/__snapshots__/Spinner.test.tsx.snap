// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render Spinner component correctly with props 1`] = `
<DocumentFragment>
  <div
    class="inset-0 z-50 "
  >
    <div
      aria-label="Loading"
      class="inline-flex flex-col gap-2 items-center justify-center absolute inset-0 m-auto z-60"
    >
      <div
        class="relative flex w-10 h-10"
      >
        <i
          class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-r-transparent border-[5px] border-b-primary-100 border-l-primary-100"
        />
        <i
          class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-r-transparent border-[5px] border-b-primary-100 border-l-primary-100"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Spinner Component should render Spinner component correctly without props 1`] = `
<DocumentFragment>
  <div
    class="inset-0 z-50 "
  >
    <div
      aria-label="Loading"
      class="inline-flex flex-col gap-2 items-center justify-center absolute inset-0 m-auto z-60"
    >
      <div
        class="relative flex w-10 h-10"
      >
        <i
          class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-r-transparent border-[5px] border-b-primary-100 border-l-primary-100"
        />
        <i
          class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-r-transparent border-[5px] border-b-primary-100 border-l-primary-100"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
